program-runtime/src/loaded_programs.rs

```diff
@@ -819,11 +819,27 @@ impl<FG: ForkGraph> ProgramCache<FG> {
             &entry.program,
             ProgramCacheEntryType::DelayVisibility
         ));
+        fn is_current_env(
+            environments: &ProgramRuntimeEnvironments,
+            env_opt: Option<&ProgramRuntimeEnvironment>,
+        ) -> bool {
+            env_opt
+                .map(|env| {
+                    Arc::ptr_eq(env, &environments.program_runtime_v1)
+                        || Arc::ptr_eq(env, &environments.program_runtime_v2)
+                })
+                .unwrap_or(true)
+        }
         let slot_versions = &mut self.entries.entry(key).or_default().slot_versions;
         match slot_versions.binary_search_by(|at| {
             at.effective_slot
                 .cmp(&entry.effective_slot)
                 .then(at.deployment_slot.cmp(&entry.deployment_slot))
+                .then(
```

Comment:
This additional `then()` has no affect on the normal operation outside the epoch boundary recompilation phase. `is_current_env()` will return `true`, so this evaluates to `.then(true.cmp(&true))` which is `.then(std::cmp::Ordering::Equal)` which is the same as no `.then()` statement as it was before.

Reply:
Same here, need some comment in the code that explains this.